xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -1449,6 +1769,12 @@ private void notifyWatcher(ResourceWatcher watcher, ResourceUpdate update) {
           throw new AssertionError("should never be here");
       }
     }
+
+    public void updateMetadataErrorState(
```

Comment:
I was going for something in between C++ `void XdsClient::UpdateResourceMetadataWithFailedParseResult()` and go `UpdateErrorMetadata()`. 

`onRejected()` sounds good, with one caveat. Do you think it may imply that the watchers will be called? Analogues to `onData()`, `onAbsent()`, `onError()`? If that's not a concern, I'll rename it.

Reply:
When a response is NACKed, no watcher should be notified, it should just behave as if never received such a response.

Reply:
Sorry for the misunderstanding. What I meant to say is—Is this a concern that `onRejected()` looks similar to the rest of `on*` methods, but, unlike the rest of them, it does not call watchers? And the users of this API may incorrectly assume that `onRejected()` will call watchers too, similar to `onData()`, `onAbsent()`, `onError()`?

Reply:
This (ResourceSubscriber) is completely internal for the XdsClient's implementation, so nothing should be worrisome as long as implementors are comfortable with it.